### PR TITLE
Fix artist name

### PR DESCRIPTION
**Pull request by gorgobacka**
At the moment, whipper always uses 'Artist in MusicBrainz' for track artist and album artist. This can lead for example to '[unknown]' in filenames.

This PR fix this issue (also mentioned here: https://github.com/JoeLametta/whipper/issues/155) and uses the artist-credit -> name if it exists (and therefor the name is different from 'Artist name in MusicBrainz'), and otherwise artist-credit -> artist -> name.

### CODE REVIEW

**Comment by Freso** [comment]
It would be great with a test case to both show case the issue and also document that your fix is working.

**Comment by gorgobacka** [comment]
@Freso Thanks a lot for your comment. I will try to include a test case over the weekend.

**Comment by gorgobacka** [comment]
I rebased the code unto latest master and added a test case to demonstrate that the metadata is set correctly for both situations. However, the test does not check the filenames.

@Freso Do you think a test to check the filenames is also needed? I expect it's not, since the variable artist (instead of sort-name) is used in the filenames.

File: whipper/test/test_common_mbngs.py [hunk 1]
```diff
@@ -114,3 +114,40 @@ def testMalaInCuba(self):
                           ';ec07a209-55ff-4084-bc41-9d4d1764e075'
                           ';f626b92e-07b1-4a19-ad13-c09d690db66c'
                           )
+
+    def testNorthernGateway(self):
+        # disc with artists tagged with [unknown] and an alias in MusicBrainz
+        # see https://github.com/JoeLametta/whipper/issues/155
```
Comment (Freso) [inline whipper/test/test_common_mbngs.py]: Make this a [docstring](http://www.pythonforbeginners.com/basics/python-docstrings/) instead of a comment. :)

Comment (gorgobacka) [inline whipper/test/test_common_mbngs.py]: Thanks for your suggestions. I will change it soon.

Comment (JoeLametta) [inline whipper/test/test_common_mbngs.py]: Looking forward to merge it when it's completed.

**Comment by gorgobacka** [comment]
I expect my previous changing needs some explanation. During my recent tests, I detected that sometimes the artist name in the recoding is different from the artist name on the release. I thought this also fits to this PR and added a fix for it. Before, whipper used always the artist name of the recording. However, this is not always correct. Now it uses the artist name of the release first. 

@Freso Can you give some insights about the artists named in recordings, please? Do you think, they are handled correctly now? What about track titles? I expect they can be also different (this might be a separate PR).

**Comment by JoeLametta** [comment]
Let's hear what @Freso thinks about it: my knowledge regarding to MusicBrainz's inner details is quite limited.

**Comment by Freso** [comment]
A Track title can be different than a Recording title (and IMHO, whipper should use the Track titles).

In the same vein, I feel whipper should use the Track artist credit and disregard the Recording artist credit entirely. (Though possibly as a switch for classical - in MusicBrainz, classical releases have the composer as the Track artist and the performers as the Recording artist(s).)

**Comment by JoeLametta** [comment]
@gorgobacka Hi, Freso replied. What's your status update about this pull request?

**Comment by gorgobacka** [comment]
@JoeLametta It can be merged. The Track Title can be addressed in a different PR. Same for a switch for classical releases.

Note: At the moment I'm still using the Recording artist as fallback, if no Track artist is present in the json file.

**Comment by RecursiveForest** [comment]
This looks good to me, although I can't say I know much about the semantics of MusicBrainz entries. I'm happy to have this merged to fix issue 155 and to address using track title & using a classical switch in another PR. If I don't hear a NAK from @JoeLametta, @Freso, or @Wizzup in the next few days I'll press the button.

**Comment by gorgobacka** [comment]
@RecursiveForest, @JoeLametta If there is nothing missing, can you please merge this PR?

**Comment by JoeLametta** [comment]
> If there is nothing missing, can you please merge this PR?

Merged, thanks.

**Comment by gorgobacka** [comment]
Thanks.

**Pull request merged**